src/Vision/testpi.py

```diff
@@ -0,0 +1,378 @@
+
+        self.__find_contours_input = self.hsl_threshold_output
+        self.__find_contours_external_only = False
+
+        self.find_contours_output = None
+
+    def process(self, source0):
+        """
+        Runs the pipeline and sets all outputs to new values.
+        """
+        # Step HSL_Threshold0:
+        self.__hsl_threshold_input = source0
+        (self.hsl_threshold_output) = self.__hsl_threshold(self.__hsl_threshold_input, self.__hsl_threshold_hue, self.__hsl_threshold_saturation, self.__hsl_threshold_luminance)
+
+        # Step Find_Contours0:
+        self.__find_contours_input = self.hsl_threshold_output
+        (self.find_contours_output) = self.__find_contours(self.__find_contours_input, self.__find_contours_external_only)
+
+    @staticmethod
+    def __hsl_threshold(input, hue, sat, lum):
+        """Segment an image based on hue, saturation, and luminance ranges.
+        Args:
+            input: A BGR numpy.ndarray.
+            hue: A list of two numbers the are the min and max hue.
+            sat: A list of two numbers the are the min and max saturation.
+            lum: A list of two numbers the are the min and max luminance.
+        Returns:
+            A black and white numpy.ndarray.
+        """
+        out = cv2.cvtColor(input, cv2.COLOR_BGR2HLS)
+        return cv2.inRange(out, (hue[0], lum[0], sat[0]), (hue[1], lum[1], sat[1]))
+
+    @staticmethod
+    def __find_contours(input, external_only):
+        """Sets the values of pixels in a binary image to their distance to the nearest black pixel.
+        Args:
+            input: A numpy.ndarray.
+            external_only: A boolean. If true only external contours are found.
+        Return:
+            A list of numpy.ndarray where each one represents a contour.
+        """
+        if(external_only):
+            mode = cv2.RETR_EXTERNAL
+        else:
+            mode = cv2.RETR_LIST
+        method = cv2.CHAIN_APPROX_SIMPLE
+        im2, contours, hierarchy = cv2.findContours(input, mode=mode, method=method)
+        return contours
+
+
+configFile = "/boot/frc.json"
+
+
+class CameraConfig: pass
+
+
+rightAngle = -15
+leftAngle = -75
+deadzone = 15
+
+team = None
+server = False
+width = None
+height = None
+imageMidpoint = None
+
+cameraConfigs = []
+
+"""Report parse error."""
+
+
+def parseError(str):
+    print("config error in '" + configFile + "': " + str, file=sys.stderr)
+
+
+"""Read single camera configuration."""
+
+
+def readCameraConfig(config):
+    cam = CameraConfig()
+    global width
+    global height
+    global imageMidpoint
+
+    # name
+    try:
+        cam.name = config["name"]
+    except KeyError:
+        parseError("could not read camera name")
+        return False
+
+    # path
+    try:
+        cam.path = config["path"]
+    except KeyError:
+        parseError("camera '{}': could not read path".format(cam.name))
+        return False
+
+    try:
+        width = config["width"]
```

Comment:
Should these be part of the camera config?

Reply:
Sad.... again.....